File: pset1 2018/mario/more/mario.c
#include <stdio.h>
#include <cs50.h>

int main(void)
{
  int size;
  // Keep prompting the user for a pyramid size until it is a valid size
  do
    {
      size = get_int("Height: ");
    }
  while (size < 0 || size > 23);

  // Main loop for pyramid
  for (int i = 1; i <= size; ++i)
    {
      // Loop for left side spaces
      for (int j = 0; j < size - i; ++j)
        {
	  printf(" ");
        }
      // Loop for left side hashes
      for (int j = 0; j < i; ++j)
        {
	  printf("#");
        }
      // Separating spaces
      printf("  ");
      // Right side hashes
      for (int j = 0; j < i; ++j)
        {
	  printf("#");
        }
      printf("\n");
    }
}

File: pset6/mario/more/mario.py
from cs50 import get_int

# Get a height from the user
while True:
    h = get_int("Height: ")
    if h >= 1 and h <= 8:
        break

# Number of lines
for i in range(h):
    # Number of spaces padded to the front
    for j in range(h - i - 1):
        print(" ", end="")
    # Number of hashes at the left
    for j in range(i + 1):
        print("#", end="")
    # Two spaces in the middle
    print("  ", end="")
    # Number of hashes at the right
    for j in range(i + 1):
        print("#", end="")
    print("")
